packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -259,14 +259,15 @@ class AndroidDevice extends Device {
     return true;
   }
 
-  Future<int> _forwardPort(String service, int devicePort, int port) async {
+  Future<Uri> _forwardUriToNewPort(String service, Uri deviceUri, int port) async {
     try {
       // Set up port forwarding for observatory.
-      port = await portForwarder.forward(devicePort, hostPort: port);
-      printTrace('$service listening on http://127.0.0.1:$port');
-      return port;
+      port = await portForwarder.forward(deviceUri.port, hostPort: port);
+      Uri localUri = deviceUri.replace(port: port);
+      printTrace('$service listening on $localUri');
```

Comment:
This trace should probably read:

'Forwarded host port $hostPort to device port $devicePort'